xdsl/transforms/eqsat_create_eclasses.py

```diff
@@ -0,0 +1,79 @@
+from xdsl.context import MLContext
+from xdsl.dialects import builtin, eqsat, func
+from xdsl.ir import Block
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.rewriter import InsertPoint, Rewriter
+from xdsl.utils.exceptions import DiagnosticException
+
+
+def insert_eclass_ops(block: Block):
+    # Insert eqsat.eclass for each operation
+    for op in block.ops:
+        results = op.results
+        if len(results) != 1:
+            raise DiagnosticException("Ops with non-single results not handled")
```

Comment:
```suggestion
        results = op.results
        if not results:
            # No need to convert ops with no results
            continue
        if len(results) != 1:
            raise DiagnosticException("Ops with non-single results not handled")
```

Reply:
This seems to handle the no results case

Reply:
It is safe to ignore return ops, but I feel that there is a need to handle other ops with no result. I meant to flag that so we know this is something needing more work. Ignoring it might cause problem when dealing with other cases such as `memref.store`, but I am not sure. Shall we skip return op only and keep `DiagnosticException` for other ops with `len(results) != 1`? i.e.:

```python
if len(results) != 1 and not isinstance(op, func.ReturnOp):
    raise DiagnosticException("Ops with non-single results not handled")
```
Happy to commit your version as well if we plan to address it later.

Reply:
Sounds good to me! MLIR has traits that might be helpful for this, like `ReturnLike`, which we might want to migrate to in the future, but I have a feeling that this will be enough for now

Reply:
Cool. I will work on refactoring this this week!